.github/workflows/integration-test-upgrade-suite.yaml

```diff
@@ -32,6 +32,27 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
+        with:
+          # this might remove tools that are actually needed,
+          # if set to "true" but frees about 6 GB
+          tool-cache: true
+
+          # all of these default to true, but feel free to set to
+          # "false" if necessary for your workflow
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          docker-images: true
+          swap-storage: true
+
+      - name: Setup tmate session
```

Comment:
Could we make this change in #12579 instead of here?

Reply:
Yes, I have mentioned in above comment that pr you mentioned should be merged before this and after that I'll rebase on top of this